transaction/core/src/input_rules.rs

```diff
@@ -53,6 +69,107 @@ impl InputRules {
                 return Err(InputRuleError::MissingRequiredOutput);
             }
         }
+
+        self.verify_partial_fill_rules(block_version, tx)?;
+
+        Ok(())
+    }
+
+    // Partial-Fill rules verification (MCIP #42)
+    fn verify_partial_fill_rules(
+        &self,
+        _block_version: BlockVersion,
+        tx: &Tx,
+    ) -> Result<(), InputRuleError> {
+        if let Some(fractional_change) = self.fractional_change.as_ref() {
+            // There is a fractional change output. Let's try to unblind its amount.
+            let fractional_change_amount = fractional_change.reveal_amount()?;
+            // Let's check if there is a corresponding real change output.
+            let real_change = tx
+                .prefix
+                .outputs
+                .iter()
+                .find(|x| fractional_change.tx_out.matches_ignoring_amount(x))
+                .ok_or(InputRuleError::MissingRealChangeOutput)?;
+            // Let's try to unblind its amount.
+            let real_change_amount =
+                try_reveal_amount(real_change, fractional_change.amount_shared_secret.as_ref())?;
+
+            // Check the bounds of the real change amount
+            if real_change_amount.token_id != fractional_change_amount.token_id {
+                return Err(InputRuleError::RealOutputTokenIdMismatch);
+            }
+            if real_change_amount.value > fractional_change_amount.value {
+                return Err(InputRuleError::RealOutputAmountExceedsFractional);
+            }
+            if self.max_allowed_change_value != 0
+                && real_change_amount.value > self.max_allowed_change_value
```

Comment:
This logic enforces minimum order volume, but the reader of the code has to perform one additional mental step to understand this- i.e subtract the `real_change_amount_value` from the `fractional_change_amount_value`. This point was discussed in the PR review video, but I'd like to +1 that point and suggest that we consider using a `minimum_order_volume_price` instead of `max_allowed_change_value` and add an extra line that does the subtraction. 

I take the point that this simplifies the logic for the transaction builder (and saves the line of code that performs the subtraction), but it's less readable IMO. That said, I know you've thought about this a lot and other people seem to not have a problem with it, so I'm fine to leave it as is. If that's the case though, I'd recommend adding a comment that includes an Alice / Bob scenario and describes the subtraction and how it computes minimum order volume.

Reply:
I think it's a good point and Nick had a similar comment, I think you guys are right and we should change this